config/dev/vsphere-credentials.yaml

```diff
@@ -24,10 +28,76 @@ kind: Credential
 metadata:
   name: vsphere-cluster-identity-cred
   namespace: ${NAMESPACE}
+  labels:
+    k0rdent.mirantis.com/component: "kcm"
 spec:
   description: vSphere credentials
   identityRef:
     apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
     kind: VSphereClusterIdentity
     name: vsphere-cluster-identity
     namespace: ${NAMESPACE}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: vsphere-cluster-identity-resource-template
+  namespace: ${NAMESPACE}
+  labels:
+    k0rdent.mirantis.com/component: "kcm"
+  annotations:
+    projectsveltos.io/template: "true"
+data:
+  configmap.yaml: |
+    {{- $$cluster := .InfrastructureProvider -}}
+    {{- $$identity := (getResource "InfrastructureProviderIdentity") -}}
+    {{- $$secret := (getResource "InfrastructureProviderIdentitySecret") -}}
+    ---
+    apiVersion: v1
+    kind: Secret
+    metadata:
```

Comment:
the component label is missing here. i am not sure whether sveltos will properly recreate everything and in the correct moment that's why i'd put the label here to avoid potential inconsistencies

Reply:
Sveltos reconciles that object, this is 100%, basically Sveltos is creating this object in the first place

Reply:
ok, please resolve the same comments if they make no sense in terms of the management cluster. i was just trying to be over conscious just in case

Reply:
resolving, keep one above for reference